src/Evolution/Systems/Cce/BoundaryData.cpp

```diff
@@ -99,6 +100,334 @@ void cartesian_to_spherical_coordinates_and_jacobians(
   get<2, 2>(*inverse_cartesian_to_spherical_jacobian) = 0.0;
 }
 
+void cartesian_spatial_metric_and_derivatives_from_spherepack(
+    const gsl::not_null<tnsr::ii<DataVector, 3>*> cartesian_spatial_metric,
+    const gsl::not_null<tnsr::II<DataVector, 3>*> inverse_spatial_metric,
+    const gsl::not_null<tnsr::ijj<DataVector, 3>*> d_cartesian_spatial_metric,
+    const gsl::not_null<tnsr::ii<DataVector, 3>*> dt_cartesian_spatial_metric,
+    const gsl::not_null<Scalar<SpinWeighted<ComplexDataVector, 0>>*>
+        derivative_buffer,
+    const gsl::not_null<Scalar<SpinWeighted<ComplexDataVector, 1>>*> eth_buffer,
+    const gsl::not_null<std::vector<double>*> interpolation_buffer,
+    const tnsr::ii<DataVector, 3>& spatial_metric_coefficients,
+    const tnsr::ii<DataVector, 3>& dr_spatial_metric_coefficients,
+    const tnsr::ii<DataVector, 3>& dt_spatial_metric_coefficients,
+    const inverse_jacobian_tensor& inverse_cartesian_to_spherical_jacobian,
+    const tnsr::I<DataVector, 3>& cartesian_coords,
+    const YlmSpherepack& spherical_harmonics,
+    const YlmSpherepack::InterpolationInfo& interpolation_info,
+    const size_t l_max, const bool radial_renormalize) noexcept {
+  const size_t size = get<0, 0>(inverse_cartesian_to_spherical_jacobian).size();
+  destructive_resize_components(cartesian_spatial_metric, size);
+  destructive_resize_components(d_cartesian_spatial_metric, size);
+  destructive_resize_components(dt_cartesian_spatial_metric, size);
+
+  destructive_resize_components(derivative_buffer, size);
+  destructive_resize_components(eth_buffer, size);
+  if (UNLIKELY(interpolation_buffer->size() != size)) {
+    *interpolation_buffer = std::vector<double>(size);
+  }
+
+  // Allocation
+  Variables<tmpl::list<::Tags::Tempijj<0, 3>, ::Tags::Tempii<0, 3>,
+                       ::Tags::TempScalar<0>>>
+      aggregated_buffers{size};
+
+  tnsr::ii<DataVector, 3>& dr_cartesian_spatial_metric =
+      get<::Tags::Tempii<0, 3>>(aggregated_buffers);
+  tnsr::ijj<DataVector, 3>& angular_d_cartesian_spatial_metric =
```

Comment:
define an mixed angular - cartesian tensor alias (i.e. separately specify dim and Frame of each index)
(do the same for `angular_d_null_l`)